code/game/gamemodes/setupgame.dm

```diff
@@ -120,7 +120,7 @@
 		if(G.block)
 			if(G.block in blocks_assigned)
 				warning("DNA2: Gene [G.name] trying to use already-assigned block [G.block] (used by [english_list(blocks_assigned[G.block])])")
-			dna_genes.Add(G)
+			dna_genes[G.type] = G

```

Comment:
call me silly but i was under the impression that you could only do associative lists with strings as the keys in byond, atleast that's how i've always seen them done.